tests/integration/test_app_sa_injection.py

```diff
@@ -0,0 +1,92 @@
+from .common import random_str
+from .test_catalog import wait_for_template_to_be_created
+import time
+from .conftest import kubernetes_api_client
+import kubernetes
+
+
+def test_app_mysql(admin_pc, admin_mc):
+    client = admin_pc.client
+    name = random_str()
+
+    ns = admin_pc.cluster.client.create_namespace(name=random_str(),
+                                                  projectId=admin_pc.
+                                                  project.id)
+    wait_for_template_to_be_created(admin_mc.client, "library")
+    answers = {
+        "defaultImage": "true",
+        "image": "mysql",
+        "imageTag": "5.7.14",
+        "mysqlDatabase": "admin",
+        "mysqlPassword": "",
+        "mysqlUser": "admin",
+        "persistence.enabled": "false",
+        "persistence.size": "8Gi",
+        "persistence.storageClass": "",
+        "service.nodePort": "",
+        "service.port": "3306",
+        "service.type": "ClusterIP"
+    }
+    testapp = client.create_app(
+        name=name,
+        externalId="catalog://?catalog=library&template=mysql&version=0.3.7&"
+                   "namespace=cattle-global-data",
+        targetNamespace=ns.name,
+        projectId=admin_pc.project.id,
+        answers=answers,
+        annotations={
+            "roleInjection.cattle.io/project-monitoring-readonly": "true",
```

Comment:
Add a test to ensure privilege escalation is not feasible by setting the roletemplate.